mock-backend/src/error/APIError.js

```diff
@@ -0,0 +1,30 @@
+class APIError extends Error {
```

Comment:
Why do you decide to use `class` approach ?
Can using plain `JavaScript` functions for that and not include some `OOP` in our backend
Try to write it in one single approach to avoid `class` etc.

Reply:
Because only classes have inheritance functionality. I need to extend Error class to add new functionality

Reply:
> Because only classes have inheritance functionality. I need to extend Error class to add new functionality

You can achieve a similar result without using `classes` by defining a set of plain JS fns that create and return error objects with the necessary properties
In JS we also can add some methods to their prototype like inheritance functionality. Not only classes have inheritance functionality. Is not it ?

Reply:
I understand, you mentioned about approach from s2s project. You want me to refactor replace class like following?

```jsx
const createError = (status, errorInfo) => {
  const err = new Error(errorInfo.message)
  err.status = status
  err.code = errorInfo.code

  return err
}

const createUnauthorizedError = () => {
  return createError(401, UNAUTHORIZED)
}

const createForbiddenError = () => {
  return createError(403, FORBIDDEN)
}

const createNotFoundError = () => {
  return createError(404, NOT_FOUND)
}

const createBadRequestError = () => {
  return createError(400, BAD_REQUEST)
}
```

We attach new values to base error object. Generally, mutating objects in javascript is not the best practice for me. If you can suggest better approach, I will be very grateful for you

Reply:
Sure. good approach! But also should pass 3 parameters in fn. `(message, status, title)` and render `message`